src/slam_toolbox_common.cpp

```diff
@@ -279,12 +455,22 @@ void SlamToolbox::publishVisualizations()
   og.info.origin.orientation.w = 1.0;
   og.header.frame_id = map_frame_;
 
-  double map_update_interval = 10;
-  map_update_interval = this->declare_parameter("map_update_interval",
-      map_update_interval);
+  double map_update_interval = this->get_parameter("map_update_interval").as_double();
   rclcpp::Rate r(1.0 / map_update_interval);
 
   while (rclcpp::ok()) {
+    {
+      // `get_current_state` is not thread safe in humble
+      // see. <https://github.com/ros2/rclcpp/issues/1746>
```

Comment:
This thread should only be running in the active state, where the thread is setup. There's no reason to check this since you setup the `join` in on deactivation to bring down the thread.

Though, this thread needs to look for that signal to join in order to exit. But this isn't necessary

Reply:
[89d4a127459a934f444459a64eb6ec60b1102c4b]

I forgot that the threads are boost::threads. I updated it to exit using boost's interruption_point.